python/ray/data/_internal/arrow_block.py

```diff
@@ -154,14 +154,15 @@ def from_bytes(cls, data: bytes) -> "ArrowBlockAccessor":
     @staticmethod
     def numpy_to_block(
         batch: Union[np.ndarray, Dict[str, np.ndarray]],
+        passthrough_arrow_not_implemented_errors: bool = False,
     ) -> "pyarrow.Table":
         import pyarrow as pa
 
         from ray.data.extensions.tensor_extension import ArrowTensorArray
 
         if isinstance(batch, np.ndarray):
             batch = {TENSOR_COLUMN_NAME: batch}
-        elif not isinstance(batch, dict) or any(
+        elif not isinstance(batch, collections.abc.Mapping) or any(
             not isinstance(col, np.ndarray) for col in batch.values()
         ):
             raise ValueError(
```

Comment:
update the error message if strict mode is set? Batch must be a dictionary of ndarrays. ndarray is no longer supported.

Reply:
I don't think we hit this path in strict mode--- I added exception type asserts to the tests to verify this.